Expensify/App
# Sync newly returned report objects from commands

### Details
We were re-fetching the chat and IOU report objects once the payIOU, payWithWallet, and rejectTransaction commands completed, which was completely unnecessary. Now, since we're returning the updated IOU and Chat report objects from https://github.com/Expensify/Web-Expensify/pull/31190, we can directly update Onyx with them.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/165863

### Tests
Make sure you have https://github.com/Expensify/Web-Expensify/pull/31318 either pulled or deployed.
1. Create accounts A and B (or use any two testing accounts with no IOU report history between one another)
2. Use A to request money from B.
3. In account B, use "I'll settle up elsewhere" to pay the IOU report.
4. Verify that the IOU report is settled, and the UI is updated to reflect it:
<img width="1792" alt="Screen Shot 2021-06-15 at 6 13 22 PM" src="https://user-images.githubusercontent.com/31285285/122035749-60419080-ce05-11eb-8e9c-49d1d9ffd84e.png">
4. Use A to request money from B.
5. In account B, decline the request.
6. Verify that the transaction was rejected correctly, and that the UI is updated to reflect it:
<img width="1792" alt="Screen Shot 2021-06-15 at 6 15 17 PM" src="https://user-images.githubusercontent.com/31285285/122035989-a5fe5900-ce05-11eb-8f04-25d242385a50.png">

### QA Steps
Same as the above tests

### Tested On

### Screenshots


https://user-images.githubusercontent.com/31285285/122041383-7a7e6d00-ce0b-11eb-98da-df9f4dcfcf47.mp4


https://user-images.githubusercontent.com/31285285/122041403-7f432100-ce0b-11eb-80b9-2ecc085305a6.mp4


https://user-images.githubusercontent.com/31285285/122041408-80744e00-ce0b-11eb-8448-b19a0db1aa59.mp4


https://user-images.githubusercontent.com/31285285/122041422-823e1180-ce0b-11eb-8956-79aee92dde3d.mp4
